Upstream-Status: Pending

Check macro OPENSSL_NO_DES before refer to openssl des related functions.
Otherwise it fails to compile if openssl disable des. It also doesn't set
default cipher function to EVP_des_ede3_cbc if the function doesn't exist.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
Index: openssl.c
===================================================================
RCS file: /cvsroot/nail/nail/openssl.c,v
retrieving revision 1.13
diff -u -r1.13 openssl.c
--- a/openssl.c	26 May 2009 21:04:15 -0000	1.13
+++ b/openssl.c	19 Sep 2017 03:22:12 -0000
@@ -688,16 +688,25 @@
 			cipher = EVP_rc2_40_cbc();
 		else if (strcmp(cp, "rc2-64") == 0)
 			cipher = EVP_rc2_64_cbc();
+#ifndef OPENSSL_NO_DES
 		else if (strcmp(cp, "des") == 0)
 			cipher = EVP_des_cbc();
 		else if (strcmp(cp, "des-ede3") == 0)
 			cipher = EVP_des_ede3_cbc();
+#endif
 		else {
 			fprintf(stderr, "Invalid cipher \"%s\".\n", cp);
 			cipher = NULL;
 		}
-	} else
+	} else {
+#ifndef OPENSSL_NO_DES
 		cipher = EVP_des_ede3_cbc();
+#else
+		fprintf(stderr, "Empty cipher.\n");
+		cipher = NULL;
+#endif
+    }
+
 	ac_free(vn);
 	return (EVP_CIPHER *)cipher;
 }
